tests/tests/regenesis.rs

```diff
@@ -0,0 +1,252 @@
+use std::{
+    ffi::OsStr,
+    net::Ipv4Addr,
+    path::PathBuf,
+    process::Stdio,
+};
+
+use fuel_core_client::client::{
+    pagination::{
+        PageDirection,
+        PaginationRequest,
+    },
+    FuelClient,
+};
+use fuel_core_types::{
+    fuel_tx::*,
+    fuel_vm::{
+        checked_transaction::IntoChecked,
+        *,
+    },
+};
+use rand::SeedableRng;
+use tempfile::{
+    tempdir,
+    TempDir,
+};
+use tokio::{
+    io::{
+        AsyncBufReadExt,
+        BufReader,
+    },
+    process::Command,
+    sync::oneshot,
+};
+
+fn workspace_root_dir() -> PathBuf {
+    let root: PathBuf = env!("CARGO_MANIFEST_DIR").into();
+    root.parent().unwrap().to_owned()
+}
+
+pub struct FuelCoreDriver {
+    pub port: u16,
+    /// This must be before the db_dir as the drop order matters here
+    pub process: tokio::process::Child,
+    pub db_dir: TempDir,
+    pub client: FuelClient,
+}
+impl FuelCoreDriver {
+    pub async fn spawn<I, S>(args: I) -> Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        // Generate temp params
+        let db_dir = tempdir().expect("Failed to create temp dir");
+        let port = portpicker::pick_unused_port().expect("Failed to pick unused port");
+
+        let mut process = Command::new(env!("CARGO"))
```

Comment:
It would be nice if, instead of starting a new child process that will compile the binary,  we used a `Command` directly from `fuel_core_bin`.

<img width="1409" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/f38d54c2-e8dc-44b8-83aa-01a6969874fd">

Like we did for snapshot tests in the https://github.com/FuelLabs/fuel-core/blob/master/bin/fuel-core/src/cli.rs#L181

Reply:
8d5f1437e